Team417_PowerPlay/src/main/java/org/firstinspires/ftc/team417_PowerPlay/InterleagueAutoLeft.java

```diff
@@ -0,0 +1,69 @@
+package org.firstinspires.ftc.team417_PowerPlay;
+
+import com.acmerobotics.roadrunner.geometry.Pose2d;
+import com.acmerobotics.roadrunner.trajectory.Trajectory;
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+
+@Autonomous(name="Interleague Left")
+public class InterleagueAutoLeft extends BaseAutonomous {
+    @Override
+    public void runOpMode() throws InterruptedException {
+        initializeAuto();
+
+        // declare trajectories here
+        Pose2d startPose = new Pose2d(0, 0, Math.toRadians(0));
+
+        Trajectory traject1 = drive.trajectoryBuilder(startPose, false)
+                .forward(5)
+                .build();
+        Trajectory traject2 = drive.trajectoryBuilder(traject1.end(), false)
+                .forward(55)
+                .build();
+        Trajectory traject3 = drive.trajectoryBuilder(traject2.end(), false)
+                .back(32)
+                .build();
+        Trajectory traject4 = drive.trajectoryBuilder(traject3.end(), false)
+                .strafeRight(16)
+                .build();
+        Trajectory traject5 = drive.trajectoryBuilder(traject4.end(), false)
+                .back(3)
+                .build();
+        Trajectory left = drive.trajectoryBuilder(traject5.end(), false)
+                .strafeLeft(48)
+                .build();
+        Trajectory middle = drive.trajectoryBuilder(traject5.end(), false)
+                .strafeLeft(18)
+                .build();
+        Trajectory right = drive.trajectoryBuilder(traject5.end(), false)
+                .strafeRight(12)
+                .build();
+
+        while (!isStarted() && !isStopRequested()) {
+            detectAprilTag();
+        }
+        updateTelemetryAfterStart();
+
+        grabberServo.setPosition(0.4);
+        drive.followTrajectory(traject1);
+        raiseAndHoldArmGroundJunctionPosition();
+
+        drive.followTrajectory(traject2);
+        drive.followTrajectory(traject3);
+
+        raiseAndHoldArmMiddleJunctionPosition();
+        drive.followTrajectory(traject4);
+        motorArm.setPower(0);
+        sleep(1300);
+        // open servo
+        grabberServo.setPosition(GRABBER_OPEN);
+        drive.followTrajectory(traject5);
+
+        if (tagOfInterest == null || tagOfInterest.id == LEFT) {
```

Comment:
This is kind of ugly. Usually you would make null be part of the else.

Reply:
When we had the null as part of the else, we got a null pointer exception for whenever it didn't see the tag because it was trying to access id, I'm not sure if there's a better way to avoid that

Reply:
Oh that makes sense. Better to leave it as is then.